internal/pkg/scorecard/interfaces.go

```diff
@@ -21,88 +21,95 @@ import (
+func (p externalPlugin) Run() scapiv1alpha1.ScorecardOutput {
 	// This should never error since we chdir in getPlugins()
-	if err := os.Chdir(viper.GetString(PluginDirOpt)); err != nil {
-		name := fmt.Sprintf("Failed Plugin: %s", filepath.Base(p.binPath))
-		description := fmt.Sprintf("Plugin with file name `%s` failed", filepath.Base(p.binPath))
+	if err := os.Chdir(filepath.Join(rootDir, scViper.GetString(PluginDirOpt))); err != nil {
+		name := fmt.Sprintf("Failed Plugin: %s", p.name)
+		description := fmt.Sprintf("Plugin with file name `%s` failed", filepath.Base(p.config.Command))
 		logs := fmt.Sprintf("failed to chdir into scorecard plugin directory: %v", err)
 		// output error to main logger as well for human-readable output
 		log.Errorf("Failed to chdir into scorecard plugin directory: %v", err)
 		return failedPlugin(name, description, logs)
 	}
-	cmd := exec.Command(p.binPath)
+	cmd := exec.Command(p.config.Command, p.config.Args...)
+	for _, env := range p.config.Env {
+		cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", env.Name, env.Value))
+	}
 	stdout := &bytes.Buffer{}
 	cmd.Stdout = stdout
 	stderr := &bytes.Buffer{}
 	cmd.Stderr = stderr
 	err := cmd.Run()
 	if err != nil {
-		name := fmt.Sprintf("Failed Plugin: %s", filepath.Base(p.binPath))
-		description := fmt.Sprintf("Plugin with file name `%s` failed", filepath.Base(p.binPath))
+		name := fmt.Sprintf("Failed Plugin: %s", p.name)
+		description := fmt.Sprintf("Plugin with file name `%s` failed", filepath.Base(p.config.Command))
 		logs := fmt.Sprintf("%s:\nStdout: %s\nStderr: %s", err, string(stdout.Bytes()), string(stderr.Bytes()))
 		// output error to main logger as well for human-readable output
-		log.Errorf("Plugin `%s` failed with error (%v)", filepath.Base(p.binPath), err)
+		log.Errorf("Plugin `%s` failed with error (%v)", p.name, err)
 		return failedPlugin(name, description, logs)
 	}
 	// parse output and add to suites
 	result := scapiv1alpha1.ScorecardOutput{}
 	err = json.Unmarshal(stdout.Bytes(), &result)
 	if err != nil {
-		name := fmt.Sprintf("Plugin output invalid: %s", filepath.Base(p.binPath))
-		description := fmt.Sprintf("Plugin with file name %s did not produce valid ScorecardOutput JSON", filepath.Base(p.binPath))
+		name := fmt.Sprintf("Plugin output invalid: %s", p.name)
+		description := fmt.Sprintf("Plugin with file name %s did not produce valid ScorecardOutput JSON", filepath.Base(p.config.Command))
 		logs := fmt.Sprintf("%s:\nStdout: %s\nStderr: %s", err, string(stdout.Bytes()), string(stderr.Bytes()))
 		// output error to main logger as well for human-readable output
-		log.Errorf("Output from plugin `%s` failed to unmarshal with error (%v)", filepath.Base(p.binPath), err)
+		log.Errorf("Output from plugin `%s` failed to unmarshal with error (%v)", p.name, err)
 		return failedPlugin(name, description, logs)
 	}
 	stderrString := string(stderr.Bytes())
 	if len(stderrString) != 0 {
-		log.Warn(stderrString)
+		log.Warnf(stderrString)
```

Comment:
Any reason for this change?

Reply:
Nope, not sure why that changed. I change it back